packages/http-client-csharp/generator/Microsoft.Generator.CSharp/src/CodeModelPlugin.cs

```diff
@@ -39,30 +40,37 @@ public CodeModelPlugin(GeneratorContext context)
         {
             Configuration = context.Configuration;
             _inputLibrary = new(() => new InputLibrary(Instance.Configuration.OutputDirectory));
+            TypeFactory = new TypeFactory();
         }
 
         // for mocking
+#pragma warning disable CS8618 // Non-nullable field must contain a non-null value when exiting constructor. Consider declaring as nullable.
         protected CodeModelPlugin()
+#pragma warning restore CS8618 // Non-nullable field must contain a non-null value when exiting constructor. Consider declaring as nullable.
         {
-            // should be mocked
-            Configuration = null!;
-            _inputLibrary = new(() => null!);
         }
 
         internal bool IsNewProject { get; set; }
 
         private Lazy<InputLibrary> _inputLibrary;
 
         // Extensibility points to be implemented by a plugin
-        public abstract TypeFactory TypeFactory { get; }
+        public virtual TypeFactory TypeFactory { get; }
         public virtual string LicenseString => string.Empty;
         public virtual OutputLibrary OutputLibrary { get; } = new();
-        public InputLibrary InputLibrary => _inputLibrary.Value;
+        public virtual InputLibrary InputLibrary => _inputLibrary.Value;
         public virtual TypeProviderWriter GetWriter(TypeProvider provider) => new(provider);
         public virtual IReadOnlyList<MetadataReference> AdditionalMetadataReferences => [];
 
         public virtual void Configure()
         {
         }
+
+        protected internal virtual IEnumerable<LibraryVisitor> GetLibraryVisitors() => _visitors;
```

Comment:
do we need both this and AddVisitor?  Are we enabling you to remove someone elses visitors?

Reply:
Yes, overriding this would allow you to skip or reorder visitors.

Reply:
I am not sure we want to do this, we can always add this later but it creates a huge risk when you start caring about the order or what came before you.

I think you choose what came before you by selecting who you derived from.